src/reponames_test.go

```diff
@@ -0,0 +1,52 @@
+package src
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_extractSourceDest(t *testing.T) {
+	src, dst, err := extractSourceDest("owner/repo")
+	require.NoError(t, err)
+	assert.Equal(t, "owner/repo", src)
+	assert.Equal(t, "owner/repo", dst)
+
+	src, dst, err = extractSourceDest("src_owner/src_repo:dst_owner/dst_repo")
+	require.NoError(t, err)
+	assert.Equal(t, "src_owner/src_repo", src)
+	assert.Equal(t, "dst_owner/dst_repo", dst)
+
+	src, dst, err = extractSourceDest("src_owner/src_repo:dst_owner/dst_repo:bogus/bogus")
+	require.Error(t, err)
+}
+
+func Test_validateNwo(t *testing.T) {
+	nwo, err := validateNwo("owner/repo")
+	require.NoError(t, err)
+	assert.Equal(t, "owner/repo", nwo)
+
+	nwo, err = validateNwo(" owner/repo ")
+	require.NoError(t, err)
+	assert.Equal(t, "owner/repo", nwo)
+
+	nwo, err = validateNwo("bogus")
+	require.Error(t, err)
+
+	nwo, err = validateNwo("bogus/")
+	require.Error(t, err)
+
+	nwo, err = validateNwo("/bogus")
+	require.Error(t, err)
+
+	nwo, err = validateNwo("bogus whitespace/bogus")
+	require.Error(t, err)
+
+	nwo, err = validateNwo("bogus/bogus/bogus")
+	require.Error(t, err)
+
+	// A separate destination is only permitted for "repo names", not NWOs.
```

Comment:
> // A separate destination is only permitted for "repo names", not NWOs.

I almost added types for `repoName` and `nwo` to make the distinction clearer, but it looked like too much of a pain to convert string slices to another type.